test/alternator/conftest.py

```diff
@@ -59,6 +59,46 @@ def pytest_collection_modifyitems(config, items):
         if "veryslow" in item.keywords:
             item.add_marker(skip_veryslow)
 
+# When testing Alternator running with --alternator-enforce-authorization=1,
+# we need to find a valid username and secret key to use in the connection.
+# Alternator allows any CQL role as the username any CQL role, and the key
+# is that role's password's salted hash. We can read a valid role/hash
+# from the appropriate system table, but can't do it with Alternator (because
+# we don't know yet the secret key!), so we need to do it with CQL.
+get_valid_alternator_role_cache = {}
```

Comment:
Why not using @cache built in decorator ?

Reply:
I never heard of it, and now I see (https://docs.python.org/3/library/functools.html) it was added in Python 3.9.
I am conflicted whether I should use it. The upside is a few ugly (but very clear) lines less. The downside is that readers (including me) would have no idea what this does under the hood, and even if it does absolutely nothing, everything will seem to work :-)

Anyway, maybe it's time that I learn something new. I'll use it :-)